File: Second_Year/DAAA(Design_and_Analysis_of_Algorithms)/comparemany.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

void minmax(const vector<int>& nos) {
    if (nos.empty()) cout << "No numbers entered" << endl;

    int minVal = nos[0];
    int maxVal = nos[0];

    for (size_t i = 1; i < nos.size(); i++) {
        if (nos[i] < minVal) minVal = nos[i];
        if (nos[i] > maxVal) maxVal = nos[i];
    }

    cout << "Min: " << minVal << ", Max: " << maxVal << endl;
}

int main() {
    vector<int> nos;
    int input;

    cout << "Enter integers (enter any non-number like 'q' to finish):" << endl;

    while (cin >> input) {
        nos.push_back(input);
    }

    if (!cin.eof()) {
        cin.clear();
        cin.ignore(numeric_limits<streamsize>::max(), '\n');
    }

    minmax(nos);
    return 0;
}

File: Second_Year/DAAA(Design_and_Analysis_of_Algorithms)/comparemany.java

import java.util.ArrayList;
import java.util.Arrays;
import java.util.Scanner;

public class comparemany {
    public static int[] comparer2(ArrayList<Integer> num) {
        if(num.isEmpty()) return new int[]{0, 0};
        int maxn = num.get(0);
        int minn = num.get(0);
        for (int i : num) {
            if(i < minn) minn = i;
            if(i > maxn) maxn = i;
        }

        return new int[]{maxn, minn};
    }

    public static void main(String[] args) {
        ArrayList<Integer> num = new ArrayList<>();
        
        System.out.print("Enter numbers and then any non-number (like 'a') to finish - ");
        try(Scanner scan = new Scanner(System.in)) {
            while(scan.hasNextInt()) {
                num.add(scan.nextInt());
            }
            scan.close();
        }
        
        int[] result = comparer2(num);
        if(Arrays.equals(result, new int[]{0, 0})) System.out.println ("No numbers entered!");
        else {
            System.out.println("Max number is " + result[0]);
            System.out.println("Min number is " + result[1]);
        }
    }
}